eng/scripts/generate_from_source_pom.py

```diff
@@ -169,6 +169,10 @@ def add_project_to_dependency_and_module_mappings(file_path: str, project_depend
         if not dependency_identifier in artifact_identifier_to_source_version:
             continue
 
+        dependency_version = get_dependency_version(dependency)
```

Comment:
@alzimmermsft do you remember why you added this exact version check? This is part of the problem that @benbp and I were seeing around the sparse checkout ordering with dev versions.  It seems like we might want to have the dependency added no matter what the version is. 

As it stands this will essentially filter out any dependency that doesn't match the current version. Perhaps that is what you were trying for though I'm just trying to better understand the reasons.